File: asistencias-main/asistencias-app/src/app/models/simulacion-aula.ts
import { SimulacionAulaItem } from "./simulacion-aula-item";

export class SimulacionAula {
    id!: number;
    nombre!: string;
    gradoId!: number;
    fechaInicio!: string;
    fechaFin!: string;
    activo!: boolean;
    cantidadSimulacion!: number;
    itemsSimulacion: SimulacionAulaItem[] = [];
}

File: estudiantes/src/main/java/com/microservice/estudiantes/models/SimulacionAula.java
package com.microservice.estudiantes.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Collection;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class SimulacionAula {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;

    private Long gradoId;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "simulacion_aula_id", nullable = false)
    private List<SimulacionAulaItem> itemsSimulacion;

    @JsonFormat(pattern = "dd-MM-yyyy HH:mm:ss")
    private LocalDateTime fechaInicio;

    @JsonFormat(pattern = "dd-MM-yyyy HH:mm:ss")
    private LocalDateTime fechaFin;

    private Boolean activo;

    private Integer cantidadSimulacion;
}
